cdap-ui/app/features/tracker/controllers/tags-ctrl.js

```diff
@@ -20,67 +20,215 @@ class TrackerTagsController{
     this.$scope = $scope;
     this.myTrackerApi = myTrackerApi;
     this.$uibModal = $uibModal;
+    this.currentPreferredPage = 1;
+    this.currentUserPage = 1;
 
-    this.tagList = {
-      preferred: 5,
-      preferredTags: [
-        {
-          'key': 'tag1',
-          'total_entity': 4
-        },
-        {
-          'key': 'tag2',
-          'total_entity': 0
-        },
-        {
-          'key': 'tag3',
-          'total_entity': 39
-        },
-        {
-          'key': 'tag4',
-          'total_entity': 17
-        },
-        {
-          'key': 'tag5',
-          'total_entity': 0
+    this.fetchTags();
+  }
+
+  fetchTags() {
+    let params = {
+      namespace: this.$state.params.namespace,
+      scope: this.$scope
+    };
+
+    this.myTrackerApi.getTags(params)
+      .$promise
+      .then((response) => {
+        this.tags = {
```

Comment:
We should be doing this initialization once when the controller load right? Do we need to do it everytime we fetch tags?